### PR TITLE
added is-cli to documentation

### PR DESCRIPTION
**Pull request by JuriBog**

### CODE REVIEW

File: documentation/docs/is-cli/is_cli_examples.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+```sh
+is create-channel -i identityA.json -t logs -s device -o channelA.json -pC test-channel
+```
+
+### Write to Channel
+
+The creator of the Channel can directly write into it.
+
+```sh
+is write-channel -i identityA.json -p "hello world" <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:42]: Change `write-channel` implementation and docs to receive channel address from a channel file (the channelA.json of the previous example). The command write-channel need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste

File: documentation/docs/is-cli/is_cli_examples.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+### Request Subscription 
+
+Other users need first to subscribe to a Channel.
+```sh
+is subscribe-channel -i identityB.json <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:49]: The command subscribe-channel need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the last parameter

File: documentation/docs/is-cli/is_cli_examples.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+### Authorize Subscription
+
+The creator of the Channel authorize one subscriber.
+```sh
+is authorize-channel -i identityA.json <Channel address of channelA> <Id of identityB>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:56]: The command authorize-channel need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the parameter

File: documentation/docs/is-cli/is_cli_examples.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+### Write message as a subscriber
+
+After the authorization the subscriber can write into the Channel.
+```sh
+is write-channel -i identityB.json -p "Can you hear me?" <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:63]: The command write-channel need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the parameter

File: documentation/docs/is-cli/is_cli_examples.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+### Read Channel
+
+The subscriber can read the Channel entries but just only the ones after he was authorized to the Channel.
+```sh
+is read-channel -i identityB.json <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:70]: The command read-channel need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the parameter

File: documentation/docs/is-cli/is_cli_examples.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+```sh
+is read-channel -i identityB.json <Channel address of channelA> 
+```
+
+### Read history of Channel
+
+The user can also read the history of a Channel. Especially if the Channel is `public` it is the only way for the subscriber to read the Channel. For that add the option `-pC` and remove the option for the preshared Key `-psK`.
+
+```sh
+is read-channel-history -i identityB.json -psK <preshared key> <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:78]: The command read-channel-history need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the parameter

File: documentation/docs/is-cli/is_cli_examples.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+### Read Channel
+
+The subscriber can read the Channel entries but just only the ones after he was authorized to the Channel.
+```sh
+is read-channel -i identityB.json <Channel address of channelA> 
+```
+
+### Read history of Channel
+
+The user can also read the history of a Channel. Especially if the Channel is `public` it is the only way for the subscriber to read the Channel. For that add the option `-pC` and remove the option for the preshared Key `-psK`.
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md]: Split that in two different example:
Read history of a public Channel
Read history of a private Channel

File: documentation/docs/is-cli/is_cli_examples.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+```sh
+is read-channel-history -i identityB.json -psK <preshared key> <Channel address of channelA> 
+```
+
+### Revoke Subscription 
+
+At any time the creator of a Channel can revoke the Subscription of other users.
+```sh
+is revoke-subscription -i identityA.json -sI <Id of identityB> <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:85]: The command revoke-subscription need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the parameter

File: documentation/docs/is-cli/is_cli_examples.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+At any time the creator of a Channel can revoke the Subscription of other users.
+```sh
+is revoke-subscription -i identityA.json -sI <Id of identityB> <Channel address of channelA> 
+```
+
+### Find subscription
+
+We can also double check if the Subscription is revoked by simply trying to find it. 
+```sh
+is find-subscription -i identityA.json -sI <Id of identityB> <Channel address of channelA>
```
Comment (mastrogiovanni) [inline documentation/docs/is-cli/is_cli_examples.md:92]: The command find-subscription need to have a -c channelFile.json where read the channel address. In this way we avoid complex copy and paste. Remove channel address as the parameter

**Review by mastrogiovanni — CHANGES_REQUESTED** [review]

File: documentation/docs/is-cli/introduction.md [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+or for one-time use
+
+```sh
+npx @iota/is-cli <command>
+```
+
+
+# Configurations
+
+### Configure CLI for local API access
```
Comment (Schereo) [inline documentation/docs/is-cli/introduction.md:40]: Maybe also add the command for configuring the CLI for usage with a single url.

File: documentation/docs/is-cli/introduction.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+
+
+# Configurations
+
+### Configure CLI for local API access 
+```sh
+is config -s http://localhost:3001 -a http://localhost:3002 -k <your-api-key>
+```
+
+### Configure API environment
```
Comment (Schereo) [inline documentation/docs/is-cli/introduction.md:45]: I would really add a little more description here. For example that this command is used for setting up the API's environment with node but only need once.
Also maybe add the other command `is setup-docker` for setting up the environment for docker.

**Comment by Schereo** [comment]
Just a general feeling of mine. I think it would be better to combine both pages into one. Less pages are better imo.

File: documentation/docs/is-cli/is_cli_examples.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
+# Examples
+
+In this example we will show how to create Identities, Channels and how to subscribe to a channel, authorize one Subscriber and write data into a Channel.
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:3]:
```suggestion
In this example we will show how to create Identities, Channels and how to subscribe to a channel, authorize a Subscriber and write data into a Channel.
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
+
+### Create json files
+
+First create two files `identityA.json` and `identitiyB.json` for creating these Identities with the CLI in the next step. Rename for the second file the `username`.
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:8]:
```suggestion
First create two files `identityA.json` and `identitiyB.json` for creating these Identities with the CLI in the next step. Rename the `username` field in `identityB.json` into `actorB`.
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+The creator of the Channel can directly write into it.
+
+```sh
+is write-channel -i identityA.json -p "hello world" -c channelA.json
+```
+
+### Request Subscription 
+
+Other users need first to subscribe to a Channel.
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:47]:
```suggestion
Other users need to subscribe to a Channel first.
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+```sh
+is subscribe-channel -i identityB.json -c channelA.json
+```
+
+### Authorize Subscription
+
+The creator of the Channel authorize one subscriber.
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:54]:
```suggestion
The creator of the Channel authorizes a subscriber.
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+```sh
+is authorize-subscription -i identityA.json -c channelA.json <Id of identityB>
+```
+
+### Write message as a subscriber
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:59]:
```suggestion
### Write Message as a Subscriber
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+
+After the authorization the subscriber can write into the Channel.
+```sh
+is write-channel -i identityB.json -p "Can you hear me?" -c channelA.json
+```
+
+### Read Channel
+
+The subscriber can read the Channel entries but just only the ones after he was authorized to the Channel.
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:68]:
```suggestion
The subscriber can read the Channel entries but only ones created after he/she was authorized to the Channel.
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+```sh
+is read-channel -i identityB.json -c channelA.json
+```
+
+### Read history of Channel
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:73]:
```suggestion
### Read History of Channel
```

File: documentation/docs/is-cli/is_cli_examples.md [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+```
+
+### Revoke Subscription 
+
+At any time the creator of a Channel can revoke the Subscription of other users.
+```sh
+is revoke-subscription -i identityA.json -c channelA.json -sI <Id of identityB>
+```
+
+### Find subscription
```
Comment (Schereo) [inline documentation/docs/is-cli/is_cli_examples.md:93]:
```suggestion
### Find Subscription
```

**Review by Schereo — APPROVED** [review]

**Review by mastrogiovanni — APPROVED** [review]

**Pull request merged**